CMakeLists.txt

```diff
@@ -57,16 +57,38 @@ set(CMAKE_CXX_VISIBILITY_PRESET hidden)
 set(CMAKE_VISIBILITY_INLINES_HIDDEN TRUE)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/bin)
 
+# Conan dependencies
+include(conan)
+set(DYNAMIC_LIB_TRUE_FALSE "False")
+if(${DYNAMIC_LIB})
+  set(DYNAMIC_LIB_TRUE_FALSE "True")
+endif()
+
+# Detect compiler, build type etc from cmake settings
+conan_cmake_autodetect(conan_settings)
+# cmake-format: off
+conan_cmake_run(
+  REQUIRES eigen/3.3.9
+           gtest/1.11.0
+           benchmark/1.5.3
```

Comment:
How does one configure where these are sourced from exactly?

Reply:
You configure remote Conan servers:
```
conan remote add remote_name remote_url
```
The order of the remotes in conan's list determines which order they are queried to find a compatible binary. If it goes through all remotes without finding a compatible binary then, if you are using the `BUILD` option, it will go through them again looking for a recipe. Recipes are python scripts which tell conan where to download source code and how to build it.